source/MaterialXCore/Definition.cpp

```diff
@@ -20,6 +20,7 @@ const string Implementation::NODE_DEF_ATTRIBUTE = "nodedef";
 const string Implementation::FILE_ATTRIBUTE = "file";
 const string Implementation::FUNCTION_ATTRIBUTE = "function";
 const string Implementation::LANGUAGE_ATTRIBUTE = "language";
+const string Implementation::UNTYPED = "none";
```

Comment:
Since the "none" type will ultimately be supported in multiple element classes, I would recommend moving this to Types.h (perhaps just beneath MULTI_OUTPUT_TYPE_STRING), and naming it NONE_TYPE_STRING.